ditto/src/main/java/eu/xenit/testing/ditto/internal/DefaultDataSetBuilder.java

```diff
@@ -4,25 +4,37 @@
 import eu.xenit.testing.ditto.api.BootstrapConfiguration;
 import eu.xenit.testing.ditto.api.BuilderConfigurer;
 import eu.xenit.testing.ditto.api.DataSetBuilder;
-import eu.xenit.testing.ditto.api.model.Transaction;
 import eu.xenit.testing.ditto.api.TransactionCustomizer;
+import eu.xenit.testing.ditto.api.model.Transaction;
 import eu.xenit.testing.ditto.internal.DefaultTransaction.TransactionBuilder;
+import eu.xenit.testing.ditto.internal.repository.DataRepository;
+import eu.xenit.testing.ditto.internal.repository.Cursor;
 import java.util.LinkedList;
 import java.util.function.Consumer;
 import lombok.AccessLevel;
 import lombok.Getter;
 
 public final class DefaultDataSetBuilder implements DataSetBuilder {
 
+    private final DataRepository storage;
+    private final Cursor cursor;
+
     @Getter(AccessLevel.PACKAGE)
     private RootContext context;
 
     @Getter(AccessLevel.PACKAGE)
     private LinkedList<Transaction> transactions = new LinkedList<>();
 
-    DefaultDataSetBuilder(BootstrapConfiguration config)
-    {
-        context = new RootContext(config);
+    DefaultDataSetBuilder(BootstrapConfiguration config) {
+        this(new DataRepository(), null, new RootContext(config));
+    }
+
+    DefaultDataSetBuilder(DataRepository storage, Cursor cursor, RootContext context) {
+        this.storage = storage;
+        this.cursor = cursor != null ? cursor : storage.getRootCursor();
+        this.context = context;
+
+        // TODO validate cursor belongs to this DataStorage instance ?
```

Comment:
I see some references to DataStorage in this PR, was DataRepository refactored from this?
If so, should we rename some variables as well for clarity?

Reply:
correct

Reply:
Should we rename them or not?
https://github.com/xenit-eu/ditto/pull/12/files#diff-5ed8a00c3d9a9ead3998910d4cbefae0R33
https://github.com/xenit-eu/ditto/pull/12/files#diff-c22fb83703391ea1853845150192d45eR22